controllers/pkg/giteaclient/mock_GiteaClient.go

```diff
@@ -0,0 +1,423 @@
+// Code generated by mockery v2.36.1. DO NOT EDIT.
```

Comment:
This file was generated by running the vektra/mockery container image and mounting the project base folder into the container. I ran the below podman command to generate this file from the top level of the project directory.
`podman run -v ${PWD}:/src:Z -w /src vektra/mockery:latest`.
On can treat this generated file as a black box and do not have to be versed with the contents of this file to write unit tests.